### PR TITLE
Add public access (both get and set) to the `label` of a `TreeNode`

### PR DESCRIPTION
**Pull request by davep**
As per #1396 this adds support for public access to the label of a `TreeNode`, via a `label` property. Both get and set access is provided. In addition to that main change, the constructor of `TreeNode` has been slightly modified so that the incoming label is processed via `Tree.process_label` to match the way that `TreeNode.set_label` works. Without this there was an asymmetry to how a `str` label was set.

### CODE REVIEW

**Review by rodrigogiraoserrao — APPROVED** [review]
LGTM

File: src/textual/widgets/_tree.py [hunk 1]
```diff
@@ -163,6 +163,17 @@ def toggle(self) -> None:
...
+    @property
+    def label(self) -> TextType:
+        """TextType: The label for the node."""
+        return self._label
+
+    @label.setter
+    def label(self, new_label: TextType) -> TextType:
+        """TextType: The label for the node."""
+        self.set_label(new_label)
+        return self.label
```
Comment (rodrigogiraoserrao) [inline src/textual/widgets/_tree.py:175]: By the way, what does it mean for the setter to return the new value?

Comment (davep) [reply src/textual/widgets/_tree.py:175]: Actually, come to think of it, in Python, nothing useful at all. Hangover from elsewhere (languages where assignments are also expressions).

Comment (davep) [reply src/textual/widgets/_tree.py:175]: Ditto Clipper with the likes of Class(y), for those of a particular vintage (if I'm recalling correctly).

**Pull request merged**